testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java

```diff
@@ -50,6 +54,40 @@ public enum BuildAndRunCmds {
             new String[]{"mvn", "dependency:sources", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString()},
             new String[]{IS_THIS_WINDOWS ? "target\\quarkus-runner.exe" : "./target/quarkus-runner"}
     }),
+    QUARKUS_JSON_PERF_PARSEONCE(new String[][]{
+            // TODO tune and report: https://www.graalvm.org/22.0/reference-manual/native-image/MemoryManagement/
+            new String[]{"mvn", "clean", "package", "-Pnative", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString(),
+                    "-Dquarkus.native.additional-build-args=" +
+                            ((UsedVersion.getVersion(false).compareTo(Version.create(23, 0, 0)) >= 0) ? "-H:+AllowDeprecatedBuilderClassesOnImageClasspath," : "") +
+                            "-R:MaxHeapSize=" + MX_HEAP_MB + "m," +
+                            "-H:-ParseOnce," +
+                            "-H:BuildOutputJSONFile=quarkus-json_minus-ParseOnce.json",
+                    "-Dfinal.name=quarkus-json_-ParseOnce"},
+            new String[]{"mvn", "package", "-Pnative", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString(),
+                    "-Dquarkus.native.additional-build-args=" +
+                            ((UsedVersion.getVersion(false).compareTo(Version.create(23, 0, 0)) >= 0) ? "-H:+AllowDeprecatedBuilderClassesOnImageClasspath," : "") +
+                            "-R:MaxHeapSize=" + MX_HEAP_MB + "m," +
+                            "-H:+ParseOnce," +
+                            "-H:BuildOutputJSONFile=quarkus-json_plus-ParseOnce.json",
+                    "-Dfinal.name=quarkus-json_+ParseOnce"},
+            //-XX:+UseShenandoahGC
+            //-XX:+UseSerialGC
+            //-XX:+UseG1GC
+            new String[]{"perf", "stat", "--delay", "2000", "java", "-Xlog:gc", "-XX:+UseSerialGC", "-Xmx" + MX_HEAP_MB + "m", "-jar", "target/quarkus-app/quarkus-run.jar"},
```

Comment:
While the comparison is not apples to apples, using serial gc is at least an attempt to make it fruit to fruit :)  
I understand that the SerialGC in Svm is a completely different implementation than SerialGC in HotSpot, yet it levels the filed a bit.
The processing of the test data is much faster with HotSpot than with Native and if you use G1 or Shenandoah the gap widens even more as Native is stuck with SerialGC...

Reply:
What's the purpose of these checks? Is it to see if GC is becoming slower in native? And do I understand correctly that if that's the case, it will be easier to observe this if we compare with a GC algorithm closer to what we are measuring?

Reply:
The purpose is to take HotSpot perf as a kind of a baseline and measure how native SVM perf differs.
To make sure the gap is not widening, preferably for us to be able to even pursue closing that gap.

By perf I mean a matrix of indicators we could optimize for, understanding the tradeoffs.
 - overall time spent processing a given workload is the most obvious one
 - CPU cycles
 - instructions
 - RSS
 - time spent doing GC

So while the SerialGC in SVM is a copying one and the SerialGC in HotSpot is a mark-sweep-compact one, it still makes sense to me to compare the overall time spent doing GC.

On the other hand, the intermediate GC events, partial GC, are not recorded for HotSpot as there is nothing to compare those events to in native run...

This whole thing started for me as a curious poking in `perf` and `callgrind` to observe differences between setting and not setting `ParseOnce`. So seeking runtime perf differences among native-image executables is another valid angle IMHO.

Reply:
ACK, thanks for the clarification